### PR TITLE
chore: remove parser function and fix types

### PR DESCRIPTION
**Pull request by aayushmau5**
**Description**

- Removed the parser function(Check https://github.com/asyncapi/diff/issues/22)
- Fix some type definitions

### CODE REVIEW

**Review by derberg — COMMENTED** [review]
- remove parser from dependencies
- make it clear in readme how to use diff, add note about validation and dereferencing mentioned by @magicmatatjahu . Point to the parser and also mention that user can use other tools as long as the file is at the end valid and dereferenced

**Comment by aayushmau5** [comment]
@derberg Updated the README. Please review :)

**Review by derberg — APPROVED** [review]
lgtm but please make a followup issue to provide a better readme and autogenerate API docs 🙏🏼 
example on API docs generation https://github.com/asyncapi/optimizer/pull/21

@magicmatatjahu wanna have a look?

File: package.json [hunk 1]
```diff
@@ -33,7 +33,6 @@
   },
   "homepage": "https://github.com/asyncapi/diff#readme",
   "dependencies": {
-    "@asyncapi/parser": "^1.5.1",
```
Comment (magicmatatjahu) [inline package.json:36]: You should use the parser in the tests, so please move this dep to the `devDependencies`.

Comment (aayushmau5) [reply package.json:36]: Cool, then I will fix the test in `diff.spec.ts` file

**Review by magicmatatjahu — APPROVED** [review]
🚀

**Pull request merged**